common/validator.js

```diff
@@ -0,0 +1,54 @@
+const Joi = require("joi");
+
+const contactValidation = Joi.object({
+  name: Joi.string().min(3).max(30).required(),
+
+  email: Joi.string()
+    .email({ minDomainSegments: 2, tlds: { allow: ["com", "net", "pl"] } })
+    .required(),
+
+  phone: Joi.string()
+    .regex(/^\d{3}-\d{3}-\d{3}$/)
+    .message({
+      "string.pattern.base": `Phone number must be written as 123-456-789`,
+    })
+    .required(),
+});
+
+const updateContact = Joi.object({
+  name: Joi.string().min(3).max(30).optional(),
+  email: Joi.string()
+    .email({
+      minDomainSegments: 2,
+      tlds: { allow: ["com", "net", "pl"] },
+    })
+    .optional(),
+  phone: Joi.string()
+    .regex(/^\d{3}-\d{3}-\d{3}$/)
+    .message({
+      "string.pattern.base": `Phone number must be written as 123-456-789.`,
+    })
+    .optional(),
+}).min(1);
+
+const validate = (schema, obj, next, res) => {
```

Comment:
Jako że to jest middleware, to sugerowałbym utworzyć nowy folder `src/middleware` i tam umieścić tę funkcję

Reply:
zrobione